Fix incorrect error messages due to QPluginLoader

lambdcalculus opened the pull request:
Specifying that the plugins are under the imageformats folder is required on Linux but not on Windows, for some reason.

Also removes an extraneous addition to the library paths, prolly some old thing; and improves debugging for when the libraries fail to load.

TrickyLeifa commented:
Image format plugins were already modified to be placed in the `imageformats` for both Windows and Linux. The plugin loader code should just be removed.

lambdcalculus commented:
Removed, and replaced with a check for whether the formats are valid.

Mentioned in Discord but I'll put it here too: turns out using QPluginLoader would just make it say it failed to load the plugins even though it did just fine.

TrickyLeifa commented on src/main.cpp:
```diff
@@ -46,16 +45,14 @@ int main(int argc, char *argv[])
+    qCritical() << "QApng plugin could not be loaded. Errors may occur when loading .apng files.";
   }
 
-  QPluginLoader webpPlugin("qwebp");
-  if (!webpPlugin.load())
+  if (!QImageReader::supportedImageFormats().contains("webp"))
   {
-    qCritical() << "QWebp plugin could not be loaded";
+    qCritical() << "QWebP plugin could not be loaded. Errors may occur when loading .webp files.";
```
You could write something like that to check and make it obvious:
```cpp
QStringList missing_formats{"webp", "apng", "gif"};
for (const QByteArray &i_format : QImageReader::supportedImageFormats())
{
  missing_formats.removeAll(i_format.toLower());
}

if (!missing_formats.isEmpty())
{
  call_error("Missing image formats: " + missing_formats.join(", ") + ".<br /><br />Please make sure you have installed the client properly.");
}
```
